### PR TITLE
Version api in spark (v1) (#5408)

### PR DESCRIPTION
**Pull request by sanaj7**

### CODE REVIEW

File: api/api-version-v1/src/main/java/com/thoughtworks/go/apiv1/version/VersionControllerV1.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            Spark.get("", mimeType, this::get);
+        });
+    }
+
+    public String get(Request req, Response res) throws IOException {
```
Comment (ketan) [inline api/api-version-v1/src/main/java/com/thoughtworks/go/apiv1/version/VersionControllerV1.java]: Should be called `show` — which is what the action is doing. `GET` is an http verb.

File: api/api-version-v1/src/main/java/com/thoughtworks/go/apiv1/version/VersionRepresenter.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+public class VersionRepresenter {
+
+
+    public static void toJSON(OutputWriter jsonWriter, CurrentGoCDVersion currentGoCDVersion){
+
+        jsonWriter.addLinks(outputLinkWriter -> outputLinkWriter
+                        .addLink("self",Routes.Version.BASE)
+                        .addAbsoluteLink("doc",Routes.Version.DOC));
+
+        jsonWriter.add("version",currentGoCDVersion.goVersion());
```
Comment (ketan) [inline api/api-version-v1/src/main/java/com/thoughtworks/go/apiv1/version/VersionRepresenter.java]: Please indent code :) `CMD+OPTION+L` in intellij

File: api/api-version-v1/src/test/groovy/com/thoughtworks/go/apiv1/version/VersionRepresenterTest.groovy [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  @BeforeEach
+  void setUp() {
+    currentGoCDVersion = new CurrentGoCDVersion()
+  }
+
+  @Test
+  void 'should render a version with hal representation'() {
+    def actualJson = toObjectString({ VersionRepresenter.toJSON(it, currentGoCDVersion) })
+    print(actualJson)
```
Comment (ketan) [inline api/api-version-v1/src/test/groovy/com/thoughtworks/go/apiv1/version/VersionRepresenterTest.groovy]: No print statements.

File: server/webapp/WEB-INF/rails/config/routes.rb [hunk 4]
```diff
@@ -256,8 +256,6 @@
 
       get 'dashboard', controller: :dashboard, action: :dashboard, as: :show_dashboard
 
-      match 'version', controller: :version, action: :show, as: :version, via: %w(get head)
```
Comment (ketan) [inline server/webapp/WEB-INF/rails/config/routes.rb]: I don't think the `head` was intentional. I suppose this support was removed, which is OK.

Comment (ketan) [reply server/webapp/WEB-INF/rails/config/routes.rb]: s/intentional/unintentional/

Comment (sanaj7) [reply server/webapp/WEB-INF/rails/config/routes.rb]: intentional : we don't need the meta information in this case in the http head

**Comment by sanaj7** [comment]
> LGTM. Can you make sure to squash commits and sign the CLA?

Should I squash all the commits into one ?

File: api/api-version-v1/src/test/groovy/com/thoughtworks/go/apiv1/version/VersionRepresenterTest.groovy [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import static com.thoughtworks.go.api.base.JsonUtils.toObjectString
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson
+
+class VersionRepresenterTest {
+
+  private CurrentGoCDVersion currentGoCDVersion;
+
+  @BeforeEach
+  void setUp() {
+    currentGoCDVersion = new CurrentGoCDVersion()
```
Comment (GaneshSPatil) [inline api/api-version-v1/src/test/groovy/com/thoughtworks/go/apiv1/version/VersionRepresenterTest.groovy]: This field should be mocked. This is calling the real `CurrentGoCDVersion` object, which value will keep changing with every new commit.

**Pull request merged**